rusk-wallet/src/lib/prompt.rs

```diff
@@ -265,69 +273,135 @@ pub(crate) fn choose_command(offline: bool) -> Option<PromptCommand> {
-            let gas_limit = request_gas_limit();
-            let gas_price = Some(request_gas_price());
-            Cli::Transfer {
+            let cmd = Cli::Transfer {
                 key,
-                rcvr,
-                amt,
-                gas_limit,
-                gas_price,
+                rcvr: request_rcvr_addr(),
+                amt: request_token_amt("transfer", balance),
+                gas_limit: request_gas_limit(),
+                gas_price: Some(request_gas_price()),
+            };
+            match confirm(&cmd) {
+                true => Some(cmd),
+                false => None,
             }
         }
         // Stake
         Prompt::Stake(key) => {
-            let stake_key = request_key_index("stake");
-            let amt = request_token_amt("stake", balance);
-            let gas_limit = request_gas_limit();
-            let gas_price = Some(request_gas_price());
-            Cli::Stake {
+            let cmd = Cli::Stake {
                 key,
-                stake_key,
-                amt,
-                gas_limit,
-                gas_price,
+                stake_key: request_key_index("stake"),
+                amt: request_token_amt("stake", balance),
+                gas_limit: request_gas_limit(),
+                gas_price: Some(request_gas_price()),
+            };
+            match confirm(&cmd) {
+                true => Some(cmd),
+                false => None,
             }
         }
         // Withdraw stake
         Prompt::Withdraw(key) => {
-            let stake_key = request_key_index("stake");
-            let gas_limit = request_gas_limit();
-            let gas_price = Some(request_gas_price());
-            Cli::WithdrawStake {
+            let cmd = Cli::WithdrawStake {
                 key,
-                stake_key,
-                gas_limit,
-                gas_price,
+                stake_key: request_key_index("stake"),
+                gas_limit: request_gas_limit(),
+                gas_price: Some(request_gas_price()),
+            };
+            match confirm(&cmd) {
+                true => Some(cmd),
+                false => None,
             }
         }
         // Export BLS Key Pair
-        Prompt::Export => {
-            let key = request_key_index("stake");
-            let encrypt = confirm_encryption();
-            Cli::Export {
-                key,
-                plaintext: !encrypt,
-            }
+        Prompt::Export => Some(Cli::Export {
+            key: request_key_index("stake"),
+            plaintext: !confirm_encryption(),
+        }),
+    }
+}
+
+/// Request user confirmation for a trasfer transaction
+fn confirm(cmd: &CliCommand) -> bool {
+    use CliCommand as Cli;
+    match cmd {
+        Cli::Transfer {
+            key: _,
+            rcvr,
+            amt,
+            gas_limit,
+            gas_price,
+        } => {
+            let max_fee = gas_limit * gas_price.unwrap();
```

Comment:
If this is safe as I think (because you always encapsulate in a `Some` put a comment and change `unwrap` to `expect`

Reply:
Yes it's safe! :)

Reply:
regardless, you should use `expect`

Reply:
I don't really think `expect` would change anything tbh. I don't want to bloat the code with a custom error message that we will never see.